### ISSUE TITLE
`Layout/AccessModifierIndentation` should only apply to bare access modifiers

### ISSUE THREAD

**Issue by deivid-rodriguez** [rubocop-hq/rubocop#6931]
## Expected behavior

I would expect the `Layout/AccessModifierIndentation` cop to only apply to bare access modifiers (that do not apply to a specific method like `private :foo`). I think my expectation comes from the experience of having never seen code following the recommendation this cop makes for these modifiers when using the `outdent` style. But also because otherwise, it makes it impossible to use this cop with `outdent` style in combination with `Layout/IndentationConsistency` because they contradict each other.

For example, I've seen this style in the wild

```ruby
module A
  class B
    def a
    end
    protected :a

    def b
    end

  private

    def c
    end
  end
end
```

but not this one (the one currently recommended)

```ruby
module A
  class B
    def a
    end
  protected :a

    def b
    end

  private

    def c
    end
  end
end
```

## Actual behavior

I would expect the first style to be followed, so my take would be to consider only bare access modifiers in this cop and rename it to `Layout/BareAccessModifierIndentation`. Non bare access modifiers would always be expected to have the same indentation as the method they apply to, which can be already enforced by the `Layout/IndentationConsistency` cop.

## Steps to reproduce the problem

Try to use both `Layout/AccessModifierIndentation` with `outdent` style and `Layout/IndentationConsistency` over the first code sample.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
0.67.2 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-linux)
```

**Comment by deivid-rodriguez** [comment]
I have already coded the fix for this, but I wanted to open an issue in case a different solution is preferred.
